operate/services/protocol.py

```diff
@@ -508,16 +508,15 @@ def __init__(
         self.rpc = rpc
         self.wallet = wallet
         self.contracts = contracts
-        self.chain_type = chain_type or ChainType.CUSTOM
 
-    def _patch(self) -> None:
-        """Patch contract and chain config."""
-        ChainConfigs.get(self.chain_type).rpc = self.rpc
-        if self.chain_type != ChainType.CUSTOM:
-            return
+        if chain_type is None:
+            self.chain_type = ChainType.CUSTOM
```

Comment:
I think this is not correct. What if the user specifies chain_type = ChainType.CUSTOM in the constructor?